File: src/org/usfirst/frc5293/util/preferences/IntPref.java
package org.usfirst.frc5293.util.preferences;

import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;

public class IntPref extends Pref<Integer> {
    public IntPref(String key, int defaultValue) {
        super(key, defaultValue);
    }

    @Override
    public Integer get() {
        return (int) SmartDashboard.getNumber(key, defaultValue);
    }

    @Override
    public void set(Integer value) {
        SmartDashboard.putNumber(key, value);
    }
}

File: src/main/kotlin/org/usfirst/frc5293/framework/prefs/IntPref.kt
package org.usfirst.frc5293.framework.prefs

import edu.wpi.first.wpilibj.Preferences

class IntPref(key: String, default: Int) : Pref<Int>(key, default) {
    override fun get() = Preferences.getInstance().getInt(key, default)

    override fun set(value: Int) {
        super.set(value)
        Preferences.getInstance().putInt(key, value)
    }
}
